src/client/components/setting-panel/tree-list.jsx

```diff
@@ -357,6 +360,45 @@ export default class ItemListTree extends React.PureComponent {
     )
   }
 
+  duplicateItem = (e, item) => {
+    e.stopPropagation()
+    const newbookmark = copy(item)
+    newbookmark.id = ''
+    newbookmark.title = newbookmark.title + '-copy'
+
+    const {
+      bookmarkGroups
+    } = this.props
+
+    const groupid = this.findBookmarkGroupId(bookmarkGroups, item.id)
+    this.props.store.modifier({
+      currentBookmarkGroupId: groupid
+    })
+    this.props.onClickItem(newbookmark)
```

Comment:
Duplicate should works as one click to duplicate it at once, after click duplicate button, there should be another item appears in bookmarks list, with the title `copy:{old title}` or `copy:{old host}`(if no title), then go into edit this duplicated item mode, user can edit it and save.

Reply:
“one click to duplicate it at once”, make sure it do this.

Reply:
It is only duplicate bookmark info.  But not save it until user click save button. So i think this is not problem.

Reply:
No, it is not a common practice, think about the duplicate item button in other apps. When user click it, they would expect item duplicated created.

Reply:
Your current solution is not duplicating a bookmark, it is copying item info and pasting into new bookmark form.

Reply:
> Your current solution is not duplicating a bookmark, it is copying item info and pasting into new bookmark form.

thank you for suggest. I will improve this. 
But now my macbook is bloom. So i will be improve this in this weekend

Reply:
No problem, take your time.